test/clients/openshift/client.go

```diff
@@ -20,11 +22,14 @@ import (
 	"k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/client-go/tools/clientcmd/api/latest"
 
-	fakerp "github.com/openshift/openshift-azure/pkg/fakerp/shared"
+	internalapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp/shared"
 	"github.com/openshift/openshift-azure/pkg/util/managedcluster"
+	"github.com/openshift/openshift-azure/test/clients"
+	testlogging "github.com/openshift/openshift-azure/test/util/log"
 )
 
-type Client struct {
+type client struct {
```

Comment:
The reason for the lower case was to localize it here?  Encapsulation?

Reply:
Having two structs called client, Client in this might be confusing.  Would it make sense to call this a kubeclient?

Reply:
Yes I lower-cased the previous client (change from Client to client) to prevent them from being exported from the package (and therefore being used by developers). In its stead I introduced a new `Client` which is the heart of this PR.

Reply:
> Having two structs called client, Client in this might be confusing. Would it make sense to call this a kubeclient?

Makes sense. I actually initially used `client` and `DeepTestClient`. I'm open to the idea of changing the names.

Reply:
I think this is fine. Because if to follow the naming convention and package naming we should avoid things like `openshift.KubeClient` as it is same as `openshift.OpenshiftClient` . Practice is that package name defines what it is, and names are generic. In this case `openshift.Client` make sense. Otherwise, we end up with duplications in the package name and method it exposes.

Reply:
going with `test/e2e/standard`